src/go/rpk/pkg/api/admin/admin.go

```diff
@@ -254,16 +270,53 @@ func (a *AdminAPI) sendToLeader(
 	if len(a.urls) == 1 {
 		return a.sendOne(method, path, body, into, true)
 	}
-	leaderID, err := a.GetLeaderID()
-	if err != nil {
-		return err
-	}
-	url, err := a.brokerIDToURL(*leaderID)
-	// if it's not possible to map the leaderID to a broker URL -> broadcast
-	if err != nil {
-		return a.sendAll(method, path, body, into)
+
+	retries := 3
+	var leaderID *int
+	var leaderURL string
+	for leaderID == nil || leaderURL == "" {
+		var err error
+		leaderID, err = a.GetLeaderID()
+		if _, ok := err.(*ErrNoAdminAPILeader); ok {
+			// No leader?  We might have contacted a recently-started node
+			// who doesn't know yet, or there might be an election pending,
+			// or there might be no quorum.  In any case, retry in the hopes
+			// the cluster will get out of this state.
+			retries -= 1
+			if retries == 0 {
+				return err
+			}
+			time.Sleep(noLeaderBackoff)
+		} else if err != nil {
```

Comment:
Maybe a switch can make this if/else block easier to read. WDYT?

Reply:
Hmm, I don't usually move up to a to switch unless there are more branches than this (only 3 here).

The function overall isn't particularly beautiful, I looked a bit at refactoring it but I think the complexity here is just real: we do have multiple levels of retry, the nesting reflects that.